bciers/apps/registration/app/components/operations/OperationRegistrationForm.tsx

```diff
@@ -0,0 +1,90 @@
+"use client";
+
+import { useCallback, useMemo, useState } from "react";
+import MultiStepFormBase from "@bciers/components/form/MultiStepFormBase";
+import { useParams, useRouter } from "next/navigation";
+import { RJSFSchema } from "@rjsf/utils";
+import { operationRegistrationUiSchema } from "apps/registration/app/data/jsonSchema/operationRegistration";
+import FacilityDataGrid from "apps/administration/app/components/facilities/FacilityDataGrid";
+import { FacilityInitialData } from "apps/administration/app/components/facilities/types";
+import { IChangeEvent } from "@rjsf/core";
+
+interface Props {
+  schema: RJSFSchema;
+  formData?: any;
+  facilityInitialData?: FacilityInitialData;
+}
+
+const OperationRegistrationForm = ({
+  facilityInitialData,
+  formData,
+  schema,
+}: Readonly<Props>) => {
+  const [error, setError] = useState(undefined);
+  const [formState, setFormState] = useState(formData ?? {});
+  const router = useRouter();
+  const params = useParams();
+  const formSection = parseInt(params?.formSection.toString());
+  const operationId = params?.operation.toString();
+  const formSectionList = schema.properties && Object.keys(schema.properties);
+  const isNotFinalStep = formSection !== formSectionList?.length;
+
+  const handleSubmit = async () => {
+    // This will have to be pulled from the response after the first page
+    const OPERATION_ID = "002d5a9e-32a6-4191-938c-2c02bfec592d";
+    // This will have to be pulled from the response after the second page
+    const OPERATION_NAME = "Operation name placeholder";
+
+    const nextStepUrl = `/operation/${OPERATION_ID}/${
+      formSection + 1
+    }?title=${OPERATION_NAME}`;
+
+    if (isNotFinalStep) {
+      router.push(nextStepUrl);
+    }
+  };
+
+  const isFacilityDataGrid = formSection === 3;
```

Comment:
This is what we did for previous multi step forms but this will potentially be problematic due to the conditional page in this form. I wonder if there is a better way to figure out which page we are on?